src/System.Text.Json/tests/Serialization/EnumTests.cs

```diff
@@ -385,14 +385,93 @@ public static void EnumAsUInt16Max()
         public static void EnumAsByteMax()
         {
             SimpleTestClass obj = JsonSerializer.Parse<SimpleTestClass>(s_jsonByteEnum);
-            Assert.Equal(SampleByteEnum.Max, obj.MyByteEnum);
+            Assert.Equal(SampleEnumByte.Max, obj.MyByteEnum);
         }
 
         [Fact]
         public static void EnumAsSByteMax()
         {
             SimpleTestClass obj = JsonSerializer.Parse<SimpleTestClass>(s_jsonSByteEnum);
-            Assert.Equal(SampleSByteEnum.Max, obj.MySByteEnum);
+            Assert.Equal(SampleEnumSByte.Max, obj.MySByteEnum);
+        }
+
+        [Theory]
+        [InlineData(sbyte.MinValue, SampleEnumSByte.MinNegative)]
+        [InlineData(short.MinValue, SampleEnumInt16.MinNegative)]
+        [InlineData(int.MinValue, SampleEnumInt32.MinNegative)]
+        [InlineData(long.MinValue, SampleEnumInt64.MinNegative)]
+        [InlineData(0, SampleEnum.MinZero)]
+        [InlineData(0, SampleEnumByte.MinZero)]
+        [InlineData(0, SampleEnumUInt16.MinZero)]
+        [InlineData(0, SampleEnumUInt32.MinZero)]
+        [InlineData(0, SampleEnumUInt64.MinZero)]
+        [InlineData(0, SampleEnumSByte.Zero)]
+        [InlineData(0, SampleEnumInt16.Zero)]
+        [InlineData(0, SampleEnumInt32.Zero)]
+        [InlineData(0, SampleEnumInt64.Zero)]
+        [InlineData(1, SampleEnum.One)]
+        [InlineData(1, SampleEnumByte.One)]
+        [InlineData(1, SampleEnumSByte.One)]
+        [InlineData(1, SampleEnumInt16.One)]
+        [InlineData(1, SampleEnumUInt16.One)]
+        [InlineData(1, SampleEnumInt32.One)]
+        [InlineData(1, SampleEnumUInt32.One)]
+        [InlineData(1, SampleEnumInt64.One)]
+        [InlineData(1, SampleEnumUInt64.One)]
+        [InlineData(2, SampleEnum.Two)]
+        [InlineData(2, SampleEnumByte.Two)]
+        [InlineData(2, SampleEnumSByte.Two)]
+        [InlineData(2, SampleEnumInt16.Two)]
+        [InlineData(2, SampleEnumUInt16.Two)]
+        [InlineData(2, SampleEnumInt32.Two)]
+        [InlineData(2, SampleEnumUInt32.Two)]
+        [InlineData(2, SampleEnumInt64.Two)]
+        [InlineData(2, SampleEnumUInt64.Two)]
+        [InlineData(byte.MaxValue, SampleEnumByte.Max)]
+        [InlineData(sbyte.MaxValue, SampleEnumSByte.Max)]
+        [InlineData(short.MaxValue, SampleEnumInt16.Max)]
+        [InlineData(ushort.MaxValue, SampleEnumUInt16.Max)]
+        [InlineData(int.MaxValue, SampleEnumInt32.Max)]
+        [InlineData(uint.MaxValue, SampleEnumUInt32.Max)]
+        [InlineData(long.MaxValue, SampleEnumInt64.Max)]
+        [InlineData(ulong.MaxValue, SampleEnumUInt64.Max)]
+        public static void ToString_Max(object expected, object enumValue)
```

Comment:
nit: Given this is testing a lot of different enum values (and not just "Max"), we may want to rename the test.

Reply:
Will cover in a next PR